### ISSUE TITLE
Request: Add Makefile, for unix-like systems native compatibility.

### ISSUE THREAD

**Issue by EndlessEden** [OpenSAGE/OpenSAGE#507]
I know this may be a bit early to ask this, but as im attempting to make a package for downstream, digging through the (lack of) compiling/installation instructions, and attempting to deduce how msbuild works, the lack of a Makefile build system is really hitting me.

Makefile doesnt replace the VS build system, it just streamlines the building and testing structure to make it easier for end-users/testers to build and install your code. - the msbuild application is a nightmare to try and automate effectively, and relies heavily on GUI elements(or poorly documented switches) to control build options. It also ultimately doesnt provide a easy way to extract the necessary libraries and executables from the source file-structure post-compilation to make usuable native system packages from.

While developer-distributed binaries are helpful for windows and (often) macos systems where the supporting libraries rarely change, unix-like operating systems, particularly linux, system libraries are often in flux. So developer-distributed binaries may pose some unique challenges or even result in licensing issues, by exclusively relying on those distributed binaries. 

---

As far as implementing a Makefile, it is rather simple, and another related sister project (openra), has already done this in the past - (https://github.com/OpenRA/OpenRA/blob/bleed/Makefile)

Using this as a reference would go a very long ways to quickly implementing support without much effort, and some one familiar with the msbuild system and file-structure could have a Makefile draft most certianly in minutes. (i would write one and push it myself if understood the build system it better)

So please consider implementing a Makefile.

**Comment by Tarcontar** [comment]
Hi, 
as you said I dont think that it makes a lot of sense at this stage of the project. As long as we do not have any active developers using linux this will get outdated soon and thus wont be of much use.

**Comment by dginovker** [comment]
@EndlessEden As a Linux-user, I would say you're better off writing a short script that compiles OpenSAGE or distributing the binary directly.

I got OpenSAGE compiled for Linux using the [developer-guide](https://github.com/OpenSAGE/OpenSAGE/blob/master/docs/developer-guide.md), but heads up it immediately crashes due to a graphics bug.

**Comment by feliwir** [comment]
I think the **dotnet** cli command is quite mature and works fine for now. As @dginovker  mentioned we still have dozens of other bugs on Linux that we (or some new contributor) needs to work out before we can think about package maintainers or similar target audiences. Once the project reaches a stable & mature state it probably makes sense though.

Thanks for your input anyways, i'll keep the issue open for now

**Comment by EndlessEden** [comment]
@dginovker Thanks  for the heads-up on the developer guide. - i will take a look later at it, and see how i can utilize it. Thank you for your advice.
